state-chain/pallets/cf-elections/src/electoral_systems/oracle_price/state_machine.rs

```diff
@@ -252,6 +252,13 @@ impl<T: OPTypes> ExternalChainStates<T> {
 				)
 			})
 	}
+
+	pub fn get_latest_asset_state(&self, asset: T::AssetPair) -> Option<AssetState<T>> {
+		all::<ExternalPriceChain>()
+			.filter_map(|chain| self[chain].price.get(&asset))
+			.max_by_key(|price_state| price_state.timestamp.median)
+			.cloned()
+	}
```

Comment:
Actually this is better than the `get_latest_price`, we should replace all occurrences of it with this. I can do the change.